### PR TITLE
Remove suppress_immediate_execution in OrderSet::IssueOrder.

### PR DESCRIPTION
**Pull request by LGM-Doyle**
Replace it with single execution on the client and server of each order.

This means that if an order needs to be issued during game load it can
take immediate effect on the client, and still not be re-executed by the
MapWnd initialization.

This does not need a special bool parameter.  It just works regardless of whether the order is coming from a loaded order or an order issued before or after the `MapInit()` function.

### CODE REVIEW

File: util/Order.cpp [hunk 1]
```diff
@@ -34,12 +34,19 @@ Empire* Order::GetValidatedEmpire() const {
...
     ExecuteImpl();
     m_executed = true;
 }
 
-bool Order::Undo() const
-{ return UndoImpl(); }
+bool Order::Undo() const {
+    auto undone =  UndoImpl();
+    if (undone)
+        m_executed = false;
```
Comment (geoffthemedio) [inline util/Order.cpp:47]: Is this a good idea? If rescinding an order, it probably shouldn't be re-executed when loading...

File: util/SerializeOrderSet.cpp [hunk 2]
```diff
@@ -34,8 +36,15 @@ BOOST_CLASS_EXPORT(ForgetOrder)
 template <class Archive>
 void Order::serialize(Archive& ar, const unsigned int version)
 {
-    ar  & BOOST_SERIALIZATION_NVP(m_empire)
-        & BOOST_SERIALIZATION_NVP(m_executed);
+    ar  & BOOST_SERIALIZATION_NVP(m_empire);
+    /* m_executed is intentionally not serialized so that an order is run once
```
Comment (geoffthemedio) [inline util/SerializeOrderSet.cpp]: discuss how this works when loading a game? ie. all orders re-executed when a game is loaded, assuming the saved gamestate is that before any orders were executed by the client?

File: client/human/HumanClientApp.cpp [hunk 3]
```diff
@@ -1016,13 +1016,13 @@ bool HumanClientApp::ToggleFullscreen() {
     return true;
 }
 
-void HumanClientApp::StartGame() {
+void HumanClientApp::StartGame(bool is_new_game) {
```
Comment (geoffthemedio) [inline client/human/HumanClientApp.cpp:1023]: seems unrelated, though I suppose isn't. a comment indicating the relevance might be helpful.

**Comment by geoffthemedio** [comment]
Seems to work OK. Is there a particular test case, or are you fairly sure this will work as intended?

File: client/human/HumanClientApp.cpp (hunk 3 above)
Comment (LGM-Doyle) [reply client/human/HumanClientApp.cpp:1023]: I added a comment that `is_new_game` is false for loaded games.

**Comment by LGM-Doyle** [comment]
There is a simple test case, which demonstrates the problem and exercises the new code.
- Quickstart a game.
- On turn 1 create a new ship design.
- save the game before you hit next turn
- load the just saved game.
- the log will have several error messages of the form `Tried to remove a ShipDesign id = 0that the empire wasn't remembering`.  Other than the error messages there is no other problem.  This PR results in no error message, because the code "knows" that the game is being reloaded on turn 1 and does not add a second set of orders to create the saved ship designs for the human empire.  (The formatting of the error message is already fixed in https://github.com/freeorion/freeorion/commit/a1bb05e438350af16a7a9581c868bae4a4de5620)

File: client/human/HumanClientApp.cpp (hunk 3 above)
Comment (geoffthemedio) [reply client/human/HumanClientApp.cpp:1023]: "is" or "should be" ?

Comment (LGM-Doyle) [inline client/human/HumanClientApp.cpp]: Changed to should and rebased,

Comment (LGM-Doyle) [reply client/human/HumanClientApp.cpp:1023]: Changed to should and rebased.
